docs/getting_started/kernel.md

```diff
@@ -0,0 +1,347 @@
+
+[embedmd]:# (../../pkg/kernel/module_options.go /type ModuleOption func/ /\)/)
+```go
+type ModuleOption func(ms *ModuleConfig)
+```
+
+A _ModuleOption_ is a function that has access to the internal configuration values of a module. They are used for configuring how a kernel should run a given module.
+
+[embedmd]:# (../../pkg/kernel/module.go /type \(/ /\n\)/)
+```go
+type (
+	ModuleFactory      func(ctx context.Context, config cfg.Config, logger log.Logger) (Module, error)
+	MultiModuleFactory func(ctx context.Context, config cfg.Config, logger log.Logger) (map[string]ModuleFactory, error)
+)
+```
+
+A _ModuleFactory_ is a function returning a _Module_, while a _MultiModuleFactory_ returns a map with _ModuleFactory_ values.
+
+More details about modules and module options can be found [_here_](getting_started/module.md).
+
+### Adding modules to a _Kernel_
+
+Having seen those data types, we can now take a closer look at the three _Add_ methods from the _Kernel_ interface:
+
+- `Add(name string, moduleFactory ModuleFactory, opts ...ModuleOption)`
+  * Notice that while a Module contains functionality to be run, we cannot add a module directly to a _Kernel_, we need instead to add a _ModuleFactory_. This is because using a _ModuleFactory_ allows passing of initial configuration from the kernel to the module.
+- `AddFactory(factory MultiModuleFactory)`
+  * The _AddFactory_ method takes in a _MultiModuleFactory_ which knows how to produce a map where the keys are names and the values are _ModuleFactories_. There are some differences between using one `AddFactory` call versus issuing multiple `Add` calls:
+    * all _ModuleFactories_ produced by that one _MultiModuleFactory_ will be run using identical _ModuleOptions_
+    * the actual _MultiModuleFactory_ function call will be invoked when  the kernel starts, not before
+- `AddMiddleware(middleware Middleware, position Position)`
+  * _AddMiddleware_ gives us the option to configure a chain of _Middleware_ option with which to wrap every _Module_ functionality. Currently, the two _Position_ options available are `"beginning"` and `"end"`. The handlers added be the first two methods will run in parallel once the _Kernel_ starts, each handler wrapped in a chain of middleware.
+
+### Kernel options
+
+Another method offered by _Kernel_ objects is:
+
+[embedmd]:# (../../pkg/kernel/kernel.go /func \(k \*kernel\) Option/ /{/)
+```go
+func (k *kernel) Option(options ...Option) error {
+```
+
+_Option_ allows us to add configuration options to a kernel object, returning an error if that kernel has already started running, or if there was an error in parsing that _Option_. While useful, it is not part of the _Kernel_ interface. Notice these are _kernel.Option_ not _kernel.ModuleOption_.
+
+[embedmd]:# (../../pkg/kernel/kernel.go /type Option func/ /\n/)
+```go
+type Option func(k *kernel) error
+```
+
+The two most important predefined options are:
```

Comment:
I still don't get why the timeout for kernel shutdown (used only very sparingly in our backend) or even something only needed for tests are the most important options. Are all the other options that useless? Why do we even need a valuation here? Maybe this should just be two example options